lib/pf/Switch/Extreme/EXOS.pm

```diff
@@ -0,0 +1,189 @@
+
+use strict;
+use warnings;
+use Net::SNMP;
+use base ('pf::Switch::Extreme::Summit');
+
+# importing switch constants
+use pf::Switch::constants;
+use pf::util;
+use pf::log;
+use pf::constants;
+use pf::config qw(
+    $WEBAUTH_WIRED
+    $WIRED_MAC_AUTH
+    $WIRED_802_1X
+);
+use pf::SwitchSupports qw(
+    RoleBasedEnforcement
+    ExternalPortal
+);
+
+sub description { "Extreme EXOS" } 
+
+sub returnRoleAttribute { "Filter-Id" }
+
+=head2 findIfdescUsingSNMP
+
+Calls the switch to obtain the interface description of an ifindex
+
+=cut
+
+sub findIfdescUsingSNMP {
+    my ($self, $ifIndex) = @_;
+    my $logger = get_logger;
+    my $oid_ifDesc = '1.3.6.1.2.1.2.2.1.2';
+    if ( !$self->connectRead() ) {
+        return 0;
+    }
+    $logger->trace(
+        "SNMP get_request for ifOperStatus: $oid_ifDesc.$ifIndex");
+    my $result = $self->cachedSNMPRequest([-varbindlist => ["$oid_ifDesc.$ifIndex"]], {expires_in => '24h'});
+    return $result->{"$oid_ifDesc.$ifIndex"};
+}
+
+=head2 parseRequest
+
+Parse the RADIUS request, overriding here to fetch the ifDesc using SNMP if it can't be extracted from the packet
+
+=cut
+
+sub parseRequest {
+    my ($self, $radius_request) = @_;
+    my ($nas_port_type, $eap_type, $mac, $port, $user_name, $nas_port_id, $session_id, $ifDesc) = $self->SUPER::parseRequest($radius_request);
+    $ifDesc = $ifDesc || $self->findIfdescUsingSNMP($port);
+    return ($nas_port_type, $eap_type, $mac, $port, $user_name, $nas_port_id, $session_id, $ifDesc);
+}
+
+=head2 parseExternalPortalRequest
+
+Parse an external portal request
+
+=cut
+
+sub parseExternalPortalRequest {
+    my ( $self, $r, $req ) = @_;
+    my $logger = $self->logger;
+
+    # Using a hash to contain external portal parameters
+    my %params = ();
+
+    my $client_mac = clean_mac($req->param('mac'));
+
+    my $locationlog = pf::locationlog::locationlog_view_open_mac($client_mac);
+    my $switch_id = $locationlog->{switch};
+    my $client_ip = defined($r->headers_in->{'X-Forwarded-For'}) ? $r->headers_in->{'X-Forwarded-For'} : $r->connection->remote_ip;
+
+    my $redirect_url;
+    if ( defined($req->param('dest')) ) {
+        $redirect_url = $req->param('dest');
+    }
+    elsif ( defined($r->headers_in->{'Referer'}) ) {
+        $redirect_url = $r->headers_in->{'Referer'};
+    }
+
+    %params = (
+        switch_id               => $switch_id,
+        client_mac              => $client_mac,
+        client_ip               => $client_ip,
+        redirect_url            => $redirect_url,
+        synchronize_locationlog => $FALSE,
+        connection_type         => $WEBAUTH_WIRED,
+    );
+
+    return \%params;
+}
+
+
+=item wiredeauthTechniques
+
+Returns the reference to the deauth technique or the default deauth technique.
+
+=cut
+
+sub wiredeauthTechniques {
+    my ($self, $method, $connection_type) = @_;
+    my $logger = $self->logger;
+    if ($connection_type == $WIRED_802_1X) {
+        my $default = $SNMP::SNMP;
```

Comment:
Why not $SNMP::RADIUS ?

Reply:
SNMP::RADIUS doesn't work as well during my tests when mixing 802.1x and MAB

It seems it does MAB for a bit after the deauth, then 802.1x kicks back in after a minute or so

With up/down, 802.1x has priority like it should